tests/xdc/history_replication_signals_and_updates_test.go

```diff
@@ -117,99 +151,373 @@ func (s *historyReplicationConflictTestSuite) SetupSuite() {
+	s.HistoryRequire.EqualHistoryEventsAndVersions(fmt.Sprintf(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	3 WorkflowTaskStarted
+	4 WorkflowTaskCompleted
+	5 WorkflowExecutionUpdateAccepted {"ProtocolInstanceId": "%s", "AcceptedRequest": {"Input": {"Args": {"Payloads": [{"Data": "\"cluster2-update-input\""}]}}}}
+	`, cluster2UpdateId), []int{1, 1, 2, 2, 2}, t.cluster1.getHistory(ctx))
+
+	// cluster2 has reapplied the accepted update from cluster 1 on top of its own update, changing it from state
+	// Accepted to state Admitted, since it must be submitted to the validator on the new branch.
+	s.HistoryRequire.EqualHistoryEventsAndVersions(fmt.Sprintf(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	3 WorkflowTaskStarted
+	4 WorkflowTaskCompleted
+	5 WorkflowExecutionUpdateAccepted {"ProtocolInstanceId": "%s", "AcceptedRequest": {"Input": {"Args": {"Payloads": [{"Data": "\"cluster2-update-input\""}]}}}}
+	6 WorkflowExecutionUpdateAdmitted {"Request": {"Meta": {"UpdateId": "%s"}, "Input": {"Args": {"Payloads": [{"Data": "\"cluster1-update-input\""}]}}}}
+	7 WorkflowTaskScheduled
+	`, cluster2UpdateId, cluster1UpdateId), []int{1, 1, 2, 2, 2, 2, 2}, t.cluster2.getHistory(ctx))
+
+	// Cluster2 sends the reapplied update to cluster1, bringing the cluster histories into agreement.
+	t.cluster2.executeHistoryReplicationTasksFromClusterUntil(enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ADMITTED)
+	s.EqualValues(t.cluster1.getHistory(ctx), t.cluster2.getHistory(ctx))
+
+	s.NoError(t.cluster2.pollAndCompleteUpdate(cluster2UpdateId))
+	s.HistoryRequire.EqualHistoryEventsAndVersions(fmt.Sprintf(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	3 WorkflowTaskStarted
+	4 WorkflowTaskCompleted
+	5 WorkflowExecutionUpdateAccepted {"ProtocolInstanceId": "%s", "AcceptedRequest": {"Input": {"Args": {"Payloads": [{"Data": "\"cluster2-update-input\""}]}}}}
+	6 WorkflowExecutionUpdateAdmitted {"Request": {"Meta": {"UpdateId": "%s"}, "Input": {"Args": {"Payloads": [{"Data": "\"cluster1-update-input\""}]}}}}
+	7 WorkflowTaskScheduled
+	8 WorkflowTaskStarted
+	9 WorkflowTaskCompleted
+   10 WorkflowExecutionUpdateCompleted {"Meta": {"UpdateId": "%s"}}
+  `, cluster2UpdateId, cluster1UpdateId, cluster2UpdateId), []int{1, 1, 2, 2, 2, 2, 2, 2, 2, 2}, t.cluster2.getHistory(ctx))
+}
+
+// TestConflictResolutionReappliesUpdatesSameIds creates a split-brain scenario in which both clusters believe they are
+// active. Both clusters then accept an update and write it to their own history, but those updates have the same update
+// ID.
+func (s *hrsuTestSuite) TestConflictResolutionReappliesUpdatesSameIds() {
+	// TODO (dan) The event-reapply implementation should drop the update since its update ID conflicts with an
+	// in-flight update. Currently this is not done and hence replication fails with "Update ID update-id is already
+	// present in mutable state"
+	s.T().SkipNow()
```

Comment:
I'll address this in my next PR; we need to wire through the update registry from higher in the stack to the event-reapply implementation, so that we can check for in-flight updates with conflicting IDs.